### PR TITLE
stabilize ptr.is_aligned, move ptr.is_aligned_to to a new feature gate

### PR DESCRIPTION
**Pull request by Gankra**
This is an alternative to #121920

### CODE REVIEW

File: compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch [hunk 1]
```diff
@@ -27,6 +27,6 @@ index d0a119c..76fdece 100644
  mod slice;
  mod str;
  mod str_lossy;
--- 
+--
 2.42.1
```
Comment (Gankra) [inline compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch:31]: hmm text editor settings, not sure if this breaks something

**Comment by m-ou-se** [comment]
FCP to stabilize `is_aligned` while keeping `is_aligned_to` unstable:

@rfcbot merge

We might want to [remove](https://github.com/rust-lang/rust/pull/121920) `is_aligned_to`, but let's leave that discussion separate from stabilizing `is_aligned`.

**Review by dtolnay — APPROVED** [review]
Thank you!

**Comment by Gankra** [comment]
rebased

File: compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch (hunk 1 above)
Comment (workingjubilee) [reply compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch:31]: Does it still work with the appropriate `git apply`?

Comment (Gankra) [reply compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch:31]: how would i test that? i have no idea how these are used.

Comment (workingjubilee) [reply compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch:31]: just standing in the repo and slapping in `git apply compiler/rustc_codegen_gcc/patches/libgccjit12/0001-core-Disable-portable-simd-test.patch`

checked and it does, this is gtg. :+1:

**Comment by workingjubilee** [comment]
Looks fine now!

@bors r=dtolnay

**Comment by WaffleLapkin** [comment]
@Gankra since this merged and you reused the tracking issue number, could you update #96284 to accurately represent what the issue tracks (and tracked)?

**Comment by workingjubilee** [comment]
I fixed up the issue~

**Pull request merged**